code/game/objects/items/devices/tamagotchi.dm

```diff
@@ -0,0 +1,187 @@
+	base_icon_state = "default"
+	slot_flags = ITEM_SLOT_BELT
+	w_class = WEIGHT_CLASS_SMALL
+	force = 2
+	throwforce = 2
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	inhand_icon_state = "electronic"
+	custom_materials = list(/datum/material/iron=SMALL_MATERIAL_AMOUNT, /datum/material/glass=SMALL_MATERIAL_AMOUNT, /datum/material/plastic=SMALL_MATERIAL_AMOUNT)
+	// interaction_flags_atom = parent_type::interaction_flags_atom | INTERACT_ATOM_ALLOW_USER_LOCATION | INTERACT_ATOM_IGNORE_MOBILITY
+	var/list/icons_available = list()
+	var/radial_icon_file = 'icons/hud/radial_tamagotchi.dmi'
+
+	var/hunger = 0       // Hunger level (0 = full, 100 = starving)
+	var/happiness = 100    // Happiness level (0 = sad, 100 = very happy)
+	var/energy = 100       // Energy level (0 = tired, 100 = full of energy)
+	var/age = 0            // Age in "days" or some unit of time
+	var/alive = 0          // Status flag (1 = alive, 0 = dead)
+	var/died = 0		   // Status flag (1 = died, 0 = alive)
+
+
+/obj/item/tamagotchi/Initialize(mapload)
+	. = ..()               // Call the parent constructor
+	spawn()
+	update()   // Start the update loop
+
+
+/obj/item/tamagotchi/examine(mob/user)
+	. = ..()
+	if(in_range(src, user) || isobserver(user))
+		. += "[readout()]"
+
+/obj/item/tamagotchi/proc/readout()
+	if (alive == 0 && died == 0)
+		var/status = "The Nyamagotchi is ready to be started!"
+		return span_notice(status)
+	if (alive == 0 && died == 1)
+		var/status = "The Nyamagotchi is DEAD. You're a terrible person."
+		return span_notice(status)
+	else
+		var/status = "Hunger: [hunger], Happiness: [happiness], Energy: [energy], Age: [age]"
+		return span_notice(status)
+
+/obj/item/tamagotchi/proc/update_available_icons()
+	icons_available = list()
+	if(alive == 0)
+		icons_available += list("Start!" = image(radial_icon_file,"start"))
+	else
+		icons_available += list("Feed" = image(radial_icon_file,"feed"),
+			"Play" = image(radial_icon_file,"play"),
+			"Rest" = image(radial_icon_file,"rest"),
+			"Check Status" = image(radial_icon_file,"status"))
+
+/obj/item/tamagotchi/attack_self(mob/user)
+	update_available_icons()
+	if(icons_available)
+		var/selection = show_radial_menu(user, src, icons_available, radius = 38, require_near = TRUE, tooltips = TRUE)
+		if(!selection)
+			return
+		switch(selection)
+			if("Start!")
+				start()
+			if("Feed")
+				feed()
+			if("Play")
+				play()
+			if("Rest")
+				rest()
+			if("Check Status")
+				check_status()
+
+
+/obj/item/tamagotchi/proc/start()
+	alive = 1
+	died = 0
+	say("I'm alive, nya!")
+	playsound(src, 'sound/misc/bloop.ogg', 50, FALSE)
+	spawn(1 SECONDS)
+		playsound(src, 'sound/creatures/cat/cat_meow1.ogg', 50, FALSE)
+	update()  // Start the update loop
+
+// status update loop
+/obj/item/tamagotchi/proc/update()
+	if(alive)
+		spawn(9 SECONDS)  // Wait 10 seconds between updates (adjust as needed)
```

Comment:
spawn and sleep are generally frowned upon in TG code, use callbacks/timers instead

Reply:
Okay, I'll change it!